core/src/Streamly/Internal/Data/Fold/Window.hs

```diff
@@ -351,3 +353,23 @@ mean = Fold.teeWith (/) sum length
 {-# INLINE range #-}
 range :: (Monad m, Num a, Ord a) => Fold m (a, Maybe a) a
 range = Fold.teeWith (-) maximum minimum
+
+-- | Drop repeated elements within the specified tumbling window in the stream.
+-- /Pre-release/
+--
+{-# INLINE nub #-}
+nub :: (Monad m, Ord a) => Fold m (a, Maybe a) (Set.Set a)
+nub = Fold step initial extract
+
+    where
+
+    initial =
+        return $ Partial Set.empty
+
+    step set (new, mOld) =
+        return $ Partial $
+            case mOld of
+                Just a -> Set.delete a set `seq` Set.insert new set
```

Comment:
BUG 1: the delete is ineffective as you are inserting the new element in the original set in which the element is not deleted.

BUG 2: The element can be deleted from the Set only when the last reference of that element inside the current window is removed. We talked about this earlier, I had suggested to  use a refcount and decrement the refcount when an element is removed from the window, the element can be removed from the set when the refcount reaches 0.